packages/components/src/templates/next/components/complex/CollectionWidget/utils/getCollectionPages.ts

```diff
@@ -0,0 +1,42 @@
+import type { ProcessedCollectionCardProps } from "~/interfaces"
+import type { IsomerSiteProps } from "~/types"
+import type { IsomerCollectionPageSitemap } from "~/types/sitemap"
+import {
+  COLLECTION_PAGE_DEFAULT_SORT_BY,
+  COLLECTION_PAGE_DEFAULT_SORT_DIRECTION,
+} from "~/types"
+import {
+  getCollectionItems,
+  processCollectionItems,
+} from "../../../../layouts/Collection/utils"
+
+interface GetCollectionPagesProps {
+  site: IsomerSiteProps
+  collectionParent: IsomerCollectionPageSitemap
+}
+
+export const NUMBER_OF_PAGES_TO_DISPLAY = 3
+
+export const getCollectionPages = ({
+  site,
+  collectionParent,
+}: GetCollectionPagesProps): ProcessedCollectionCardProps[] => {
+  const items = getCollectionItems({
+    site,
+    permalink: collectionParent.permalink,
+    sortBy:
+      collectionParent.collectionPagePageProps?.defaultSortBy ??
+      COLLECTION_PAGE_DEFAULT_SORT_BY,
+    sortDirection:
+      collectionParent.collectionPagePageProps?.defaultSortDirection ??
+      COLLECTION_PAGE_DEFAULT_SORT_DIRECTION,
+  })
+
+  if (items.length === 0) {
+    throw new Error(
+      `CollectionWidget: No collection items found for reference link ${collectionParent.permalink}`,
+    )
+  }
```

Comment:
just a failsafe to force codebuild failureas there's no graceful fallback and this should not happen - would rather it trigger codebuild failure alarms so that we can investigate - open to better ideas